scripts/install

```diff
@@ -1,127 +1,214 @@
+language=$YNH_APP_ARG_LANGUAGE
+disablecacheck=$YNH_APP_ARG_DISABLECACHECK
+
+# Set and store language
+language=${LANGUAGES[$language]}
 
 # Set app specific variables
-app="$YNH_APP_INSTANCE_NAME"
+app=$YNH_APP_INSTANCE_NAME
 dbname=$app
 dbuser=$app
 
-# Check domain/path availability
-sudo yunohost app checkurl "${domain}${path}" -a "$app" \
-  || exit 1
+#=================================================
+# CHECK IF THE APP CAN BE INSTALLED WITH THESE ARGS
+#=================================================
 
-# Set and store language
-language=${LANGUAGES[$3]}
-ynh_app_setting_set "$app" language "$language"
+final_path=/var/www/$app
+test ! -e "$final_path" || ynh_die "This path already contains a folder"
 
 # Define LOGDIR (create it later when user is created)
 LOGDIR=/var/log/$app
+test ! -e "$LOGDIR" || ynh_die "This log path already contains a folder"
+
+# Normalize the url path syntax
+path_url=$(ynh_normalize_url_path "$path_url")
+
+# Check web path availability
+ynh_webpath_available "$domain" "$path_url"
+# Register (book) web path
+ynh_webpath_register "$app" "$domain" "$path_url"
 
-# Check destination directory
-DESTDIR="/var/www/${app}"
-[[ -d "$DESTDIR" ]] && ynh_die \
-"The destination directory '${DESTDIR}' already exists.\
- You should safely delete it before installing this app."
 
 # Check whether Baïkal or Radicale is installed
-if sudo yunohost app list --installed -f baikal | grep -q id ; then
+baikal_install=$(agendav_app_check_installation "baikal")
+radicale_install=$(agendav_app_check_installation "radicale")
+if [[ -n "$baikal_install" ]]
+then
     caldav_app="baikal"
     caldav_baseurl="/cal.php/"
-elif sudo yunohost app list --installed -f radicale | grep -q id ; then
+elif [[ -n "$radicale_install" ]]
+then
     caldav_app="radicale"
     caldav_baseurl="/"
 else
-    ynh_die "You must install Baïkal or Radicale before"
+    ynh_die "Baikal or Radicale is mandatory"
 fi
 
+#=================================================
+# STORE SETTINGS FROM MANIFEST
+#=================================================
+
+ynh_app_setting_set "$app" language "$language"
+ynh_app_setting_set "$app" path "$path_url"
+ynh_app_setting_set "$app" domain "$domain"
+ynh_app_setting_set "$app" disablecacheck "$disablecacheck"
+ynh_app_setting_set "$app" final_path "$final_path"
+
+#=================================================
+# STANDARD MODIFICATIONS
+#=================================================
+
 # Install dependencies
-ynh_package_is_installed "php5-cli" \
-  || ynh_package_install "php5-cli"
+ynh_install_app_dependencies php5-cli
 
-# Create tmp directory and fetch app inside
-TMPDIR=$(mktemp -d)
-extract_agendav "$TMPDIR"
+#=================================================
+# CREATE A MYSQL DATABASE
+#=================================================
 
-# Generate random password and encryption key
+# Initialize database
 dbpass=$(ynh_string_random)
+ynh_app_setting_set "$app" mysqlpwd "$dbpass"
```

Comment:
These lines are useless.
Just remove the third argument ($dbpass) in the following ynh_mysql_create_db call.
It will create a new password and store it for you.